src/Database/Table/SqlBuilder.php

```diff
@@ -552,6 +552,16 @@ protected function addWhereComposition(array $columns, array $parameters)
 		}
 	}
 
+	private function getConditionHash($condition, $parameters) {
+		foreach ($parameters as &$parameter) {
+			if ($parameter instanceof Selection) {
+				$parameter = $this->getConditionHash($parameter->getSql(), $parameter->sqlBuilder->getParameters());
+			} elseif (is_object($parameter) && method_exists($parameter, '__toString')) {
```

Comment:
Primary yes, but it could be be somethink else (in our app we are using entities, for example new Product($activeRow), that old behaviour ignores too). I came from SqlPreprocessor::formatValue, where object with __toString is supported too...